tests/runtime/prelude/stdlib/other/StringIterator.sk

```diff
@@ -0,0 +1,100 @@
+module String;
+
+extension mutable class StringIterator {
+  mutable fun forward(n: Int): mutable StringIterator {
+    for (_ in Range(0, n)) _ = this.next();
+    this
+  }
+
+  mutable fun rewind(n: Int): mutable StringIterator {
+    for (_ in Range(0, n)) _ = this.prev();
+    this
+  }
+
+  mutable fun takeString(n: Int = Int::max): String {
+    String::fromChars(Array::createFromIterator(this.take(n)));
+  }
+
+  mutable fun takeUntil(end: readonly StringIterator): String {
+    String::fromGenerator(() -> {
+      while (this < end) {
+        yield this.next().fromSome()
+      }
+    })
+  }
+
+  readonly fun compare(other: readonly StringIterator): Order {
+    for (i in this.clone().zipLongest(other.clone())) {
+      cmp = i.i0.compare(i.i1);
+      if (!(cmp is EQ())) return cmp;
+    } else {
+      EQ()
+    }
+  }
+
+  // Returns true if the iterator starts with the given prefix.  If the prefix
+  // matches then the iterator is moved forward to the end of the match
+  // otherwise it is unchanged.
+  mutable fun startsWith(other: readonly StringIterator): Bool {
+    i = this.clone();
+    for (ch in other.clone()) {
+      i.next() match {
+      | Some(ch2) if (ch != ch2) -> return false
+      | Some _ -> void
+      | None() -> return return false
+      }
+    };
+    this.assign(i);
+    true
+  }
+
+  mutable fun search(pattern: String): Bool {
```

Comment:
Let's call this `contains()`, I think most people's intuition about that name is more likely to be correct re behavior and return type. 

Note that calling it "contains" does slightly differ from the behavior of "contains()" for Sequences - there, it's whether the predicate value appears as an item in the sequence - but I think Strings and Sequences are different enough that we should break with consistency. It's always possible to *view* a String as Sequence of chars, for example.

Reply:
Calling `search()` `contains()` feels weird to me - search() implies that it's moving the iterator whereas contains() implies to me that it's not (when it really is).